[jugsliao] iP

jugsliao opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the `main`
method:

```java
public class Main {
    public static void main(String[] args) {
        new Duke("tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

jugsliao committed:
Level-1

jugsliao committed:
duke with list, level 2

jugsliao committed:
level-3 commit, mark and unmark

jugsliao committed:
level-4, todo event and deadline

jugsliao committed:
level-4, todo event and deadline

jugsliao committed:
Automated Text UI Testing

jugsliao committed:
level 5, exceptions

jugsliao committed:
Level-5 change

jugsliao committed:
level 6, delete

jugsliao committed:
Task-7, included storage to harddisk

jugsliao committed:
Level-7, included storage to harddisk

jugsliao committed:
Level-8, minimal: store dates as localdate object

jugsliao committed:
Merge branch 'branch-level-8'

jugsliao committed:
A-MoreOOP

jugsliao committed:
A-Packages, created java packages

jugsliao committed:
Merge branch 'add-gradle-support'

jugsliao committed:
A-JUnit, added tests

jugsliao committed:
A-JavaDoc

jugsliao committed:
A-CodingStandard, changed indentation for switch in Command.java

jugsliao committed:
Level-9, added find feature

jugsliao committed:
Merge branch 'branch-A-JavaDoc'

jugsliao committed:
Merge tag 'A-CodingStandard'

A-CodingStandard

jugsliao committed:
fixed merge conflicts

jugsliao committed:
added more docs, fixed minor bugs

gohyongjing commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,12 @@
+package duke.exception;
+
+public class DukeException extends Exception{
```
Perhaps you should put a space before this and other curly braces,
```suggestion
public class DukeException extends Exception {
```

gohyongjing commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,83 @@
+                FileInputStream fis = new FileInputStream(this.filePath);
+                ObjectInputStream ois = new ObjectInputStream(fis);
+                Object o;
+                while (true) {
+                    try {
+                        o = ois.readObject();
+                        if (o instanceof Task) {
+                            tasks.add((Task) o);
+                        } else {
+                            System.out.println("Object is not a task");
+                        }
+                    } catch (EOFException e) {
+                        break;
+                    }
+                }
+                ois.close();
```
Perhaps this code can be abstracted out to avoid deep nesting

gohyongjing commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,83 @@
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        } else {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                System.out.println(e);
+            }
+        }
```
Perhaps the guard clause can be placed in front of the rest of the code
See: https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent

gohyongjing commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,197 @@
+        }
+    }
+
+    /**
+     * Returns false by default and true if "bye" command is executed
+     *
+     * @return false by default and true if "bye" command is executed
+     */
```
Perhaps you should add in punctuation after each line and param

gohyongjing commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,197 @@
+     * @param storage The Storage object to save the task after execution.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        CommandName cn = CommandName.valueOf(this.commandName.toUpperCase());
```
Perhaps you could consider removing the this keyword here and the rest of the code to remove unecessary noise

```suggestion
        CommandName cn = CommandName.valueOf(commandName.toUpperCase());
```

gohyongjing reviewed (commented):
I think your code is very well written. I like that there are no complicated expressions and that your code follows major coding standards and code quality guidelines, except for a few advanced rules. 👍

jugsliao committed:
fixed command find bug

jugsliao committed:
A-CheckStyle, fixed most style checks

jugsliao committed:
Level-10, implemented a very simple GUI for duke

Bentimate commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+ */
+public class Task implements Serializable {
+    private String taskName;
+    private boolean done;
```
Should this boolean be named to sound like a boolean?

Bentimate commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,75 @@
+     *
+     * @return The size of the tasks list.
+     */
+    public int size() {
```
Any reason this getter method is named size() instead of getSize()?

Bentimate reviewed (commented):
Overall very concise and easy to understand! Good work!

gekhro commented on build.gradle:
```diff
@@ -0,0 +1,61 @@
+plugins {
```
nice plugins

gekhro commented on src/test/java/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    @Test
+    public void markTest() throws DukeException {
```
could change mark and unmark methods to be under the command class instead

gekhro commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,90 @@
+ * Stores and loads tasks from a text file.
+ */
+public class Storage {
+    private String filePath;
```
good job on privatising the filepath

gekhro commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,65 @@
+     * @param input Contains the command name or other information to create command object.
+     * @return The appropriate command object that can be executed.
+     */
+    public static Command parse(String input) throws DukeException {
```
could maybe implement java enums with the switch/case method for all the different types of commands instead of using if/else statements

jugsliao committed:
fixed some style issues

jugsliao committed:
abstract logic out in Storage.java

jugsliao committed:
Remove exception statements in Command.java

jugsliao committed:
A-Assertions

Use assertions in TaskList.java to make sure the index in delete
and get is more than 0 and less than size of the list.

jugsliao committed:
Merge pull request #2 from jugsliao/branch-A-Assertions

A-Assertions

jugsliao committed:
A-CodeQuality

Parser and Command class had methods that were too long. Abstracting
out the logic for each of the command types and creating new classes for
each of them improves the overall quality of the code.

jugsliao committed:
Merge pull request #3 from jugsliao/branch-A-CodeQuality

A-CodeQuality

jugsliao committed:
Merge branch 'branch-A-Assertions' into Master

jugsliao committed:
C-FriendlierSyntax

Included more shortcuts for command types in Parser.java

jugsliao committed:
BCD-Extension

jugsliao committed:
Merge branch 'master' of https://github.com/jugsliao/ip into Master

jugsliao committed:
BCD-Extension

jugsliao committed:
BCD-Extension

jugsliao committed:
Merge pull request #4 from jugsliao/Master

Master

jugsliao committed:
Delete import statement

jugsliao committed:
Merge pull request #5 from jugsliao/Master

Delete import statement

jugsliao committed:
Change Ui message

jugsliao committed:
Merge https://github.com/jugsliao/ip

jugsliao committed:
Fix find command bug

jugsliao committed:
Fix style Checks

jugsliao committed:
A-BetterGui, made gui nicer

jugsliao committed:
Merge pull request #6 from jugsliao/branch-Level-10

A-BetterGui, made gui nicer

jugsliao committed:
Change some respond comments to be more masculine.

jugsliao committed:
Merge branch 'master' of https://github.com/jugsliao/ip

jugsliao committed:
Handle all the errors that can be anticipated

jugsliao committed:
Set title

jugsliao committed:
Fix spelling error.

jugsliao committed:
Add Ui.png

jugsliao committed:
Update Readme.md

jugsliao committed:
Update Readme.

jugsliao committed:
Update Readme.

jugsliao committed:
Update Readme

jugsliao committed:
Update Readme

jugsliao committed:
Update Readme

jugsliao committed:
Update build.gradle

jugsliao committed:
Update README.md

Commit:
Change farewell message.

Commit:
Add full stops in messages.

jugsliao committed:
Fix bug on commands which give index
